### PR TITLE
[Mirror deployments] Allow adding shadow endpoints to endpoint creation/updating API

### PR DESCRIPTION
**Pull request by tiffzhao5**
# Pull Request Summary

Allow adding shadow endpoints to endpoint creation/updating in API.
- Modify db schema
- Update LLM and model create/update endpoints to take in shadows

Follow-up PR:
- db migration

## Test Plan and Usage Guide

Finished unit tests:
- `test_model_endpoint_use_cases.py` for creation and updating
- `test_live_model_endpoint_service.py` for creation and updating

TODO unit tests:
- `test_db_model_endpoint_record_repository.py` for creating record with shadow endpoints
- `test_llm_use_cases.py` for creating and updating 

TODO: create shortcut tickets for shadow deployments

### CODE REVIEW

File: model-engine/model_engine_server/db/models/hosted_model_inference.py [hunk 1]
```diff
@@ -453,6 +453,7 @@ class Endpoint(Base):
     current_bundle = relationship("Bundle")
     owner = Column(String(SHORT_STRING))
     public_inference = Column(Boolean, default=False)
+    shadow_endpoints_ids = Column(ARRAY(Text), default=[])
```
Comment (seanshi-scale) [inline model-engine/model_engine_server/db/models/hosted_model_inference.py:456]: @song-william guess we're gonna have to figure out an endpoint migration process now heh

Comment (yunfeng-scale) [reply model-engine/model_engine_server/db/models/hosted_model_inference.py:456]: i don't think shadow needs a migration, but production variants should need a migration

Comment (tiffzhao5) [reply model-engine/model_engine_server/db/models/hosted_model_inference.py:456]: @yunfeng-scale why doesn't adding a shadows column require a migration?

Comment (yunfeng-scale) [reply model-engine/model_engine_server/db/models/hosted_model_inference.py:456]: oh sorry i mean a backfill

**Pull request closed without merging**